tuxemon/event/actions/get_player_monster.py

```diff
@@ -20,33 +29,170 @@
+                elif filter_name == "height":
+                    field = int(target.height)
+                elif filter_name == "max_hp":
+                    field = target.hp
+                elif filter_name == "current_hp":
+                    field = target.current_hp
+                elif filter_name == "armour":
+                    field = target.armour
+                elif filter_name == "dodge":
+                    field = target.dodge
+                elif filter_name == "melee":
+                    field = target.melee
+                elif filter_name == "ranged":
+                    field = target.ranged
+                elif filter_name == "speed":
+                    field = target.speed
+                extra = int(self.extra)
+                if value_name == "less_than" and bool(lt(field, extra)):
+                    self.result = True
+                elif value_name == "less_or_equal" and bool(le(field, extra)):
+                    self.result = True
+                elif value_name == "greater_than" and bool(gt(field, extra)):
+                    self.result = True
+                elif value_name == "greater_or_equal" and bool(
+                    ge(field, extra)
+                ):
+                    self.result = True
+                elif value_name == "equals" and bool(eq(field, extra)):
+                    self.result = True
+                elif value_name == "not_equals" and bool(ne(field, extra)):
+                    self.result = True
+
+        return self.result
 
     def set_var(self, menu_item: MenuItem[Monster]) -> None:
-        self.session.player.game_variables[self.variable_name] = str(
-            menu_item.game_object.instance_id.hex
+        self.choose = True
+        player = self.session.player
+        monster = menu_item.game_object
+
+        player.game_variables[self.variable_name] = str(
+            monster.instance_id.hex
         )
         self.session.client.pop_state()
 
     def start(self) -> None:
+        self.result = False
+        self.choose = False
         # pull up the monster menu so we know which one we are saving
         menu = self.session.client.push_state(MonsterMenuState())
+        menu.is_valid_entry = self.validate  # type: ignore[assignment]
         menu.on_menu_selection = self.set_var  # type: ignore[assignment]
 
     def update(self) -> None:
         try:
             self.session.client.get_state_by_name(MonsterMenuState)
         except ValueError:
+            player = self.session.player
+            if self.result and not self.choose:
+                # the player can choose, but returns
+                player.game_variables[self.variable_name] = "no_choice"
+            if not self.result:
+                # the player can't choose (eg no females in the party)
+                player.game_variables[self.variable_name] = "no_options"
             self.stop()
+            print(player.game_variables[self.variable_name])
```

Comment:
please remove this or change to `log.debug`

Reply:
You're right, I forgot to remove